### PR TITLE
Foundationdb integration

### PR DESCRIPTION
**Pull request by vendethiel**

### CODE REVIEW

File: foundationdb/datadog_checks/foundationdb/check.py [hunk 1]
```diff
@@ -24,9 +29,83 @@ def __init__(self, name, init_config, instances):
...
     def check(self, _):
+        try:
+            status = self.fdb_status_data()
+        except SubprocessOutputEmptyError as e:
+            self.service_check("foundationdb.can_connect", AgentCheck.CRITICAL, message="Did not receive a response from `status json`")
+            raise
+
+        if status[2] != 0:
+            self.service_check("foundationdb.can_connect", AgentCheck.CRITICAL, message="`fdbcli` returned non-zero error code")
+            raise ValueError("Fdbcli failed")
```
Comment (jnthn) [inline foundationdb/datadog_checks/foundationdb/check.py]: Probably doesn't want to be captialized, and maybe include the full invocation `fdbcli satus json` in the output.

File: foundationdb/datadog_checks/foundationdb/check.py [hunk 2]
```diff
@@ -24,9 +29,83 @@ def __init__(self, name, init_config, instances):
...
+            data = json.loads(status[0])
+        except Exception as e:
+            self.service_check("foundationdb.can_connect", AgentCheck.CRITICAL, message="Could not parse `status json`")
+            raise
+
+        self.check_metrics(data)
+
+    def check_metrics(self, status):
+        if not "cluster" in status:
+            raise 1
```
Comment (jnthn) [inline foundationdb/datadog_checks/foundationdb/check.py]: Hmm....this doesn't look so good.

File: foundationdb/datadog_checks/foundationdb/check.py [hunk 3]
```diff
@@ -24,9 +29,83 @@ def __init__(self, name, init_config, instances):
...
+                for k, v in workload["transactions"].items():
+                    self.maybe_gauge("foundationdb.workload.transactions." + k + ".hz", v, "hz")
+                    self.maybe_count("foundationdb.workload.transactions." + k + ".counter", v, "counter")
+
+            if "operations" in workload:
+                for k, v in workload["operations"].items():
+                    self.maybe_gauge("foundationdb.workload.operations." + k + ".hz", v, "hz")
+                    self.maybe_count("foundationdb.workload.operations." + k + ".counter", v, "counter")
+
+        print(status.keys())
```
Comment (jnthn) [inline foundationdb/datadog_checks/foundationdb/check.py]: Debug leftover?

File: foundationdb/tests/test_foundationdb.py [hunk 4]
```diff
@@ -1,15 +1,87 @@
...
+def test_integ(aggregator, instance):
     # type: (AggregatorStub, Dict[str, Any]) -> None
     check = FoundationdbCheck('foundationdb', {}, [instance])
     check.check(instance)
 
+    for metric in METRICS:
+        aggregator.assert_metric(metric)
     aggregator.assert_all_metrics_covered()
     aggregator.assert_metrics_using_metadata(get_metadata_metrics())
+    aggregator.assert_service_check("foundationdb.can_connect", AgentCheck.OK)
```
Comment (jnthn) [inline foundationdb/tests/test_foundationdb.py:87]: This relies on `fdbcli` being installed, and so is the test we could probably upgrade to use the Docker containers?

Comment (jnthn) [reply foundationdb/tests/test_foundationdb.py:87]: (Not that doing said work is a task for this PR, to be clear. Just making sure I understand correctly.)

Comment (vendethiel) [reply foundationdb/tests/test_foundationdb.py:87]: Yes, that's the idea

**Pull request merged**